### PR TITLE
Belowground Symmetric ZOI & Fixed Salinity

### PR DESCRIPTION
**Pull request by mcwimm**
Combination (child) of SymmetricZOI & FixedSalinity; no new functionalities

### CODE REVIEW

File: TreeModelLib/BelowgroundCompetition/SZoiFixedSalinity/SZoiFixedSalinity.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        super().calculateBelowgroundResources()
+        bg_factor_symmetric_ZOI = self.belowground_resources
+
+        FixedSalinity.calculateBelowgroundResources(self)
+        bg_factor_fixed_salinity = self.belowground_resources
+
+        self.belowground_resources = bg_factor_fixed_salinity * \
+                                     bg_factor_symmetric_ZOI
+        # print("ZOI: ", bg_factor_symmetric_ZOI, ", Sal: ",
+        #       bg_factor_fixed_salinity, ' All: ', self.belowground_resources)
```
Comment (jbathmann) [inline TreeModelLib/BelowgroundCompetition/SZoiFixedSalinity/SZoiFixedSalinity.py:52]: Any reason to keep this?

Comment (mcwimm) [reply TreeModelLib/BelowgroundCompetition/SZoiFixedSalinity/SZoiFixedSalinity.py:52]: Of course not :)

**Comment by mcwimm** [comment]
In general, I just thought that instead of creating new classes that are only combinations of existing classes we could build a 'merge'-class in which x below-or above-ground classes can be combined by multiplying the resource vector. What do you think @jvollhueter @jbathmann ?

**Comment by jbathmann** [comment]
Can you please rebase?

**Comment by mcwimm** [comment]
> Can you please rebase?

done.

File: TreeModelLib/BelowgroundCompetition/SZoiFixedSalinity/SZoiFixedSalinity.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        FixedSalinity.addTree(self, tree)
+
+    ## This function returns a list of the growth reduction factors of all trees.
+    #  calculated in the subsequent time step.\n
+    #  @return: np.array with $N_tree$ scalars
+    def calculateBelowgroundResources(self):
+        super().calculateBelowgroundResources()
+        bg_factor_symmetric_ZOI = self.belowground_resources
+
+        FixedSalinity.calculateBelowgroundResources(self)
```
[the remark this answers is not in the document]
Comment (jbathmann) [reply TreeModelLib/BelowgroundCompetition/SZoiFixedSalinity/SZoiFixedSalinity.py:46]: I was thinking of the GET functions which are part of all tree models. See TreeModel.py line 13. Can you use them?

**Comment by jbathmann** [comment]
> In general, I just thought that instead of creating new classes that are only combinations of existing classes we could build a 'merge'-class in which x below-or above-ground classes can be combined by multiplying the resource vector. What do you think @jvollhueter @jbathmann ?



> In general, I just thought that instead of creating new classes that are only combinations of existing classes we could build a 'merge'-class in which x below-or above-ground classes can be combined by multiplying the resource vector. What do you think @jvollhueter @jbathmann ?

I like this idea. Let's discuss how one could achieve it without to much restructuring.

**Comment by jbathmann** [comment]
:+1:

**Pull request merged**